### PR TITLE
Modular computers changes

### PR DESCRIPTION
**Pull request by Shadowlight213**
Adds some stuff requested by cheri, and changes roundstart tablet allotment.

Also adds some basic documentation because I think cheri asked me to make some, but then apparently it was about the cell thing but I had already made it so oh well.

:cl:
tweak: Service and cargo have lost roundstart tablets and engineering has gained them
tweak: you can now charge battery modules in cell chargers. Use a screwdriver on a computer to remove them. As a note, tablets and laptops can be charged directly by placing them in a security charger.
/:cl:

### CODE REVIEW

**Comment by Ultimate-Chimera** [comment]
Why remove them from cargo?

**Comment by Iamgoofball** [comment]
computers should not need to charge

**Comment by Shadowlight213** [comment]
@Ultimate-Chimera I wasn't sure what use they had for cargo. I probably would add them back in after I made a supply console program.

**Comment by Jalleo** [comment]
Yeah sounds worthless for cargo useful for engineering.

**Comment by kevinz000** [comment]
Shadowlight: Bug with programs. If the NT network is shut off, it doesn't check for it while you're in a program. It only checks for connectivity while you're trying to open up a program. As in, turn off wireless with the control program, and KEEP IT OPEN. Turn it on anytime with your wireless tablet as it won't register ass D/C'd

**Comment by Hatfish** [comment]
Would it be within the grasp of this PR to make tablets and other similar crew computers orderable from cargo?

**Comment by kevinz000** [comment]
Why does the relay use 20000W of power which is enough to power a small department?
Quantum relay more like NTnet certified powersink.

**Comment by Shadowlight213** [comment]
@Kevinz000 the programs staying open is definitely a bug. That's odd they should immediately close upon losing connection if they need ntnet and are wireless.
Also yeah I can tweak the power ratings of the relay.

**Comment by Shadowlight213** [comment]
@hatfish sure I guess.
I'll probably make a crate like maybe 5 tablets.
@iamgoofball how many credits would that cost do you think?
There's two preset variations of tablets. Cheap, which is literally the shittiest possible, and advanced, which are decent.
Also someone asked me to port the laptop vendor machine. I was hesitant on doing it as we had no economy, but I could probably use space cash for it.

**Comment by GunHog** [comment]
UNRELATED FEATURE REQUEST:
Replace the ID consoles in each head's office with the ID Console program which restricts itself based on the head's respective access. You can even use the departmental colors for the computer's screen when used!

**Comment by Shadowlight213** [comment]
@GunHog That's sorta partially implemented already. I still need to figure out a way to split it up and give each head a different variant. the departmental restrictions are implemented though. it's a var in the program.

File: code/game/machinery/cell_charger.dm [hunk 1]
```diff
@@ -91,8 +101,11 @@
 	if(!charging)
 		return
 
-	charging.loc = loc
-	user << "<span class='notice'>You telekinetically remove [charging] from [src].</span>"
+	if(B)
+		B.loc = loc
```
Comment (optimumtact) [inline code/game/machinery/cell_charger.dm]: B.forceMove(loc)

File: code/game/machinery/cell_charger.dm [hunk 2]
```diff
@@ -91,8 +101,11 @@
 	if(!charging)
 		return
 
-	charging.loc = loc
-	user << "<span class='notice'>You telekinetically remove [charging] from [src].</span>"
+	if(B)
+		B.loc = loc
+	else
+		charging.loc = loc
```
Comment (optimumtact) [inline code/game/machinery/cell_charger.dm]: same here

File: code/modules/modular_computers/documentation.dm [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/*
```
Comment (optimumtact) [inline code/modules/modular_computers/documentation.dm]: make this an markdown formatted file please. (.md)

**Comment by Core0verload** [comment]
@Shadowlight213
I want to make a pretty big rewrite of this system, based on Animus computers system. It would conflict with your PR in "you can now charge battery modules in cell chargers", and it would implement the same functionality. Can you remove any cell-related changes for now?

**Comment by kevinz000** [comment]
what the hell is animus

**Comment by Core0verload** [comment]
Russian SS13 server I actually code for.

**Comment by Shadowlight213** [comment]
@Core0verload No prob. I mainly did that because @Cheridan requested it.

File: code/modules/modular_computers/laptop_vendor.dm [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "computer_fabricator", "Personal Computer Vendor", 500, 400, state = state)
+		ui.open()
+		ui.set_autoupdate(state = 1)
+
+
+obj/machinery/lapvend/attackby(obj/item/I as obj, mob/user as mob)
+
+	if(istype(I,obj/item/stack/spacecash)
```
Comment (PKPenguin321) [inline code/modules/modular_computers/laptop_vendor.dm]: Missing a )

**Comment by kevinz000** [comment]
@Shadowlight213 
The programs are immediately closing when the network goes down now. I don't even know why it does it sometimes?

**Comment by Boredone** [comment]
As a note, the USB things don't seem to work correctly. Tried putting them on a computer, couldn't export any files, and the thing stayed in my hands, teleporting out if I removed it from the computer (With a screwdriver)

**Pull request merged**